aait/backend/group-2A/WebApi/Controllers/PostController.cs

```diff
@@ -4,69 +4,81 @@
 using MediatR;
+using Microsoft.AspNetCore.Authorization;
 using Microsoft.AspNetCore.Mvc;
 
 namespace WebApi.Controllers;
 
 [ApiController]
+[Authorize]
 [Route("api/[controller]")]
 public class PostController : ControllerBase
 {
     private readonly IMediator _mediator;
 
-    public PostController(IMediator mediator)
-    {
+    public PostController(IMediator mediator){
         _mediator = mediator;
     }
 
-    [HttpGet]
-    public async Task<IActionResult> GetNewsFeed( )
-    {
-        // token getter to be implemented
-        var command = new GetFollowingPostRequest{Id = 1};
+    [HttpGet("User/{UserId}")]
+    public async Task<IActionResult> GetUserPosts( int UserId ){
+        var command = new GetUserPostRequest(){Id = UserId};
         var posts = await _mediator.Send(command);
-
-        return Ok(posts);
-    }
-    [HttpGet("{tag}")]
-    public async Task<IActionResult> GetByTag( string tag)
-    {
-        // token getter to be implemented
-        var command = new GetByTagRequest{Tag = tag};
-        var posts = await _mediator.Send(command);
-
         return Ok(posts);
     }
     
     
-    [HttpGet("{Search}")]
-    public async Task<IActionResult> GetByContent( string search)
+    [HttpGet]
+    public async Task<IActionResult> GetPosts([FromQuery] string? search, [FromQuery] string? tag)
     {
-        // token getter to be implemented
-        var command = new GetByContenetRequest{Contenet = search};
-        var posts = await _mediator.Send(command);
 
-        return Ok(posts);
+        if (!string.IsNullOrEmpty(search))
+        {
+            var command = new GetByContenetRequest() { Contenet = search };
+            var posts = await _mediator.Send(command);
+            return Ok(posts);
+        }
+        else if (!string.IsNullOrEmpty(tag))
+        {
+            var command = new GetByTagRequest { Tag = tag };
+            var posts = await _mediator.Send(command);
+            return Ok(posts);
+        }
+        else
+        {
+            var command = new GetFollowingPostRequest{Id = int.Parse(User.FindFirst("reader").Value)};
+            var posts = await _mediator.Send(command);
+            return Ok(posts);
+        }
     }
     
+   
+    
     [HttpPost("Add")]
-    public async Task<IActionResult> AddPost(CreatePostDto createPost){
-        // token getter to be implemented
+    public async Task<IActionResult> AddPost([FromBody]CreatePostDto createPost){
+        createPost.UserId = int.Parse(User.FindFirst("reader").Value);
         var command = new CreatePostCommand {CreatePost = createPost};
         var postId = await _mediator.Send(command);
         return Ok(postId);
     }
-    [HttpPut("{PostID}")]
-    public async Task<IActionResult> UpdatePost(UpdatePostDto updatePost){
-        // token getter to be implemented
+    
+    
+    
+    [HttpPut("{PostId}")]
+    public async Task<IActionResult> UpdatePost([FromBody] UpdatePostDto updatePost, int UserId, int PostId){
```

Comment:
you don't need to pass the userId and PostId to updatePost Action , you can define them in UpdatePostDto as field.